test/gtest/ucp/test_ucp_memheap.cc

```diff
@@ -7,6 +8,134 @@
 #include "test_ucp_memheap.h"
 
 
+void test_ucp_memheap::test_nonblocking_implicit_stream_xfer(nonblocking_send_func_t send, size_t alignment)
+{
+    static const int max_iter = 300 / ucs::test_time_multiplier();
+    static const size_t size = ucs_max((size_t)rand() % (12*1024), alignment);
+    static const size_t memheap_size = max_iter * size + alignment;
+    entity *pe0 = create_entity();
+    entity *pe1 = create_entity();
+    ucs_status_t status;
+
+    ucp_mem_h memh;
+    void *memheap = NULL;
+    status = ucp_mem_map(pe1->ucph(), &memheap, memheap_size, 0, &memh);
+    ASSERT_UCS_OK(status);
+
+    memset(memheap, 0, memheap_size);
+
+    void *rkey_buffer;
+    size_t rkey_buffer_size;
+    status = ucp_rkey_pack(pe1->ucph(), memh, &rkey_buffer, &rkey_buffer_size);
+    ASSERT_UCS_OK(status);
+
+    pe0->connect(pe1);
+    pe1->connect(pe0);
+
+    ucp_rkey_h rkey;
+    status = ucp_ep_rkey_unpack(pe0->ep(), rkey_buffer, &rkey);
+    ASSERT_UCS_OK(status);
+
+    std::string expected_data[300];
+    assert (max_iter <= 300);
+
+    for (int i = 0; i < max_iter; ++i) {
```

Comment:
please add a test which actually goes to pending (i guess max_iter should be bigger), the test should fail with the current code because flush will not actually flush it.